### ISSUE TITLE
correctness of the cycles.

### ISSUE THREAD

**Issue by miha-skalic** [arthurdouillard/snapshot_ensembles#2]
Hi @arthurdouillard,

could you confirm this implementation is cycled correctly? 
What puzzles me is the following: 
lets say  `epoch=10`, and `nb_cycles=10`. From what I understand this means that at the begging the learning rate will be reset to max, but the model will save at the end of the epoch, thus the saving will be done at the worst time possible and not right before the reset.

**Comment by miha-skalic** [comment]
to elaborate:
I think line28 (https://github.com/arthurdouillard/keras-snapshot_ensembles/blob/master/snapshot.py#L28) should be 
`if epoch == 0 or (epoch + 1) % self.period != 0: return`

**Comment by arthurdouillard** [comment]
I think you're right, it should be `(epoch + 1) % self.period`.

I'll fix it when I can. Thanks! :)

**Comment by arthurdouillard** [comment]
Fixed. Thanks again.
